nomad/drainer/watch_nodes.go

```diff
@@ -179,21 +174,31 @@ func (w *nodeDrainWatcher) watch() {
 			currentNode, tracked := tracked[nodeID]
 
 			switch {
-			// If the node is tracked but not draining, untrack
+
 			case tracked && !newDraining:
+				// If the node is tracked but not draining, untrack
 				w.tracker.Remove(nodeID)
 
-				// If the node is not being tracked but is draining, track
+			case !tracked && node.UnresponsiveStatus():
```

Comment:
Since this is a noop should we even bother computing it? I suppose the compiler might elide it, but that seems awfully aggressive.

Reply:
This computation takes 1ns or less, so I feel having it clearly understandable when looking at the code is worth it. Alternatively, we could have a big old comment on the default case to detail the matchers.

Reply:
This is a switch case so we'll always take the first matching branch. Having this no-op check prevents the node from falling thru to one of the other `!tracked` cases that come later.

Reply:
Removed in 4c8b5566ab0f681df2d0cabd823467f149eb8ad2